### PR TITLE
ipq806x: refresh config

### PR DESCRIPTION
**Pull request by graysky2**
based on make kernel_menuconfig CONFIG_TARGET=generic and 5.4.48

### CODE REVIEW

**Comment by mans0n** [comment]
I ran `make kernel_oldconfig` on 5.4.40 and the result was same as this except for `CONFIG_GRO_CELLS=y`.

File: target/linux/ipq806x/config-5.4 [hunk 1]
```diff
@@ -273,7 +267,6 @@ CONFIG_KPSS_XCC=y
 CONFIG_KRAITCC=y
 CONFIG_KRAIT_CLOCKS=y
 CONFIG_KRAIT_L2_ACCESSORS=y
-CONFIG_LEDS_TRIGGER_DISK=y
```
Comment (mans0n) [inline target/linux/ipq806x/config-5.4:276]: You are reverting f93fcf8923aa. Is it intentional?

**Comment by graysky2** [comment]
@mans0n - 
Regarding the difference of `CONFIG_GRO_CELLS=y`, shouldn't you try it against 5.4.48 which was included in https://github.com/openwrt/openwrt/commit/e44e60b290fb346a4cdd5ca9b141abbcf61e80fe?

My bad reverting https://github.com/openwrt/openwrt/commit/f93fcf8923aa1f89b8190e125ba9fbb435e26045 ... I just rebased and forced push.

**Comment by adrianschmutzler** [comment]
> I ran make kernel_oldconfig on 5.4.40 and the result was same as this except for CONFIG_GRO_CELLS=y.

CONFIG_GRO_CELLS seems to depend on the environment of the person doing the refresh. I always have it, mans0n seems to never have it. For other settings, it appears to behave the opposite way.

**Comment by graysky2** [comment]
@adrianschmutzler - I can manually comment it out and rebase but I do not know if that would lead to the most general state for any build environment.  

I did not find much in the way of documentation on this particular option at least grepping for in the kernel source and docs or [here](https://cateee.net/lkddb/web-lkddb/GRO_CELLS.html).

**Comment by adrianschmutzler** [comment]
I would keep the GRO_CELLS in, I also recently merged a refresh with it to some other target.
What I'm less sure about is whether we should keep CONFIG_PLUGIN_HOSTCC.

**Comment by adrianschmutzler** [comment]
Okay, just did a refresh myself.

I get the same config with the following diff:
```
--- a/target/linux/ipq806x/config-5.4
+++ b/target/linux/ipq806x/config-5.4
@@ -267,7 +267,6 @@ CONFIG_KPSS_XCC=y
 CONFIG_KRAITCC=y
 CONFIG_KRAIT_CLOCKS=y
 CONFIG_KRAIT_L2_ACCESSORS=y
-CONFIG_LEDS_TRIGGER_DISK=y
 CONFIG_LIBFDT=y
 CONFIG_LOCK_DEBUGGING_SUPPORT=y
 CONFIG_LOCK_SPIN_ON_OWNER=y
@@ -399,7 +398,6 @@ CONFIG_PINCTRL_MSM=y
 # CONFIG_PINCTRL_SDM660 is not set
 # CONFIG_PINCTRL_SDM845 is not set
 # CONFIG_PINCTRL_SM8150 is not set
-CONFIG_PLUGIN_HOSTCC="g++"
 CONFIG_PM_OPP=y
 CONFIG_POWER_RESET=y
 CONFIG_POWER_RESET_MSM=y
```

As stated earlier, I'd drop the PLUGIN_HOSTCC. What startles me is that CONFIG_LEDS_TRIGGER_DISK does not stick. Any ideas what's the problem there?

File: target/linux/ipq806x/config-5.4 [hunk 2]
```diff
@@ -151,21 +143,20 @@ CONFIG_CRYPTO_ZSTD=y
...
 CONFIG_DTC=y
 CONFIG_DT_IDLE_STATES=y
-# CONFIG_DWMAC_DWC_QOS_ETH is not set
 # CONFIG_DWMAC_GENERIC is not set
 CONFIG_DWMAC_IPQ806X=y
 # CONFIG_DWMAC_QCOM_ETHQOS is not set
 CONFIG_DYNAMIC_DEBUG=y
 CONFIG_EDAC_ATOMIC_SCRUB=y
 CONFIG_EDAC_SUPPORT=y
+CONFIG_ETHERNET_PACKET_MANGLE=y
```
Comment (adrianschmutzler) [inline target/linux/ipq806x/config-5.4:159]: @Ansuel This was removed when you added 5.4 support, but kernel_oldconfig wants to add it back. Can you enlighten us?

**Comment by adrianschmutzler** [comment]
@Schneehuhn Can you comment on why `make kernel_oldconfig` insists on removing CONFIG_LEDS_TRIGGER_DISK?

**Comment by graysky2** [comment]
> As stated earlier, I'd drop the PLUGIN_HOSTCC. 

Done, rebased.

**Comment by Schneehuhn** [comment]
> why `make kernel_oldconfig` insists on removing CONFIG_LEDS_TRIGGER_DISK?

CONFIG_LEDS_TRIGGER_DISK depends on CONFIG_ATA. Should we add CONFIG_ATA=m?

**Comment by mans0n** [comment]
CONFIG_PLUGIN_HOSTCC might be environment-dependent, but I guess CONFIG_GRO_CELLS was not in my refresh just because I didn't update my repo regularly...

**Comment by adrianschmutzler** [comment]
> CONFIG_LEDS_TRIGGER_DISK depends on CONFIG_ATA. The disk led trigger isn't a kernel module, instead when the symbol is set disk trigger support is compiled into libata.

> Should we add CONFIG_ATA=m? That fixes the issue.

Setting =m means we would need a module. So, either we set CONFIG_ATA=y, or we move CONFIG_LEDS_TRIGGER_DISK into a module and use AddDepends/ata. Can CONFIG_LEDS_TRIGGER_DISK be built as module?

**Comment by Schneehuhn** [comment]
> Setting =m means we would need a module.

For target/ipq806x libata (that's ata-core in OpenWrt parlance) is pulled in as a module.

CONFIG_LEDS_TRIGGER_DISK is a binary flag to compile disktrigger support into libata.ko. Some ledtriggers can be built as modules, others are switches to compile ledtrigger support into other modules, this is the latter type.

**Comment by adrianschmutzler** [comment]
It seems a little undesirable to me to have the CONFIG_LEDS_TRIGGER_DISK in target config, but the ata-core stuff built as module, and defined elsewhere. The fact that the kernel removes the symbols also seems to indicate that this is not optimal.

I wonder whether it wouldn't be more tidy to put CONFIG_ATA into the target config as well?

**Comment by adrianschmutzler** [comment]
Since the CONFIG_LEDS_TRIGGER_DISK discussion is actually off-topic, I have merged the refresh now (and adjusted author field to match SOB).

I'd be happy if somebody could pick up the CONFIG_LEDS_TRIGGER_DISK topic.

**Comment by adrianschmutzler** [comment]
Another option could be to define CONFIG_LEDS_TRIGGER_DISK=y via a meta package, add the dependency for ata there, and then add the package to DEFAULT_PACKAGES for the target.

**Comment by Ansuel** [comment]
I really like the idea of the meta package for disk led

Il Dom 5 Lug 2020, 01:40 Adrian Schmutzler <notifications@github.com> ha
scritto:

**Pull request closed without merging**